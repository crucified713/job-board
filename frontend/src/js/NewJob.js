import React, { Component } from 'react';

class NewJob extends Component {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.state = {
      title: '',
      location: '',
      description: ''
    }
  }

  handleInputChange(event) {
    const target = event.target,
    value = target.value,
    name = target.name;

    if (value.trim().length !== 0) {
      this.setState({
        [name]: value
      });
    }
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.state.title.length !== '')
      this.props.toCreateNewJob(this.state);
  }

  render() {
    return (
      <>
        <form className="jobs__form" onSubmit={this.handleSubmit}>
          <h5 className="text-center"> Post New Job</h5 >
          <div className="form-group">
            <label htmlFor="job-title" className="required">Tilte</label>
            <input className="form-control" name="title" id="job-title" placeholder="eg. Administrator" onChange={this.handleInputChange} required />
          </div>

          <div className="form-group">
            <label htmlFor="job-location">Location</label>
            <input className="form-control" name="location" id="job-location" placeholder="eg. Melbourne" onChange={this.handleInputChange} />
          </div>

          <div className="form-group">
            <label htmlFor="job-location">Description</label>
            <textarea className="form-control" name="description" id="job-location" placeholder="Please type description here..." onChange={this.handleInputChange} ></textarea>
          </div>
          <div className="text-right">
            <button disabled={this.state.title.length === 0 ? 'disabled' : false} type="submit" className="btn btn-primary">Post</button>
          </div>
        </form>
      </>
    )
  }

}



export default NewJob;
