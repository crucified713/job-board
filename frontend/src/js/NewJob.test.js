import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import React from 'react';
import NewJob from './NewJob';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';


describe('New Job', function () {
  it('should renders correctly', () => {
    const tree = renderer.create(<NewJob />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should update state with changed input value', function() {
    const wrapper = shallow(<NewJob />),
      input = wrapper.find('#job-title');
    input.simulate("change", {target: {value: 'Admin', name: 'title'}})
    wrapper.update();
    expect(wrapper.instance().state.title).toBe('Admin');
  });
});