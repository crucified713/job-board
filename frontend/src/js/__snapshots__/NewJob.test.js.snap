// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Job should renders correctly 1`] = `
<form
  className="jobs__form"
  onSubmit={[Function]}
>
  <h5
    className="text-center"
  >
     Post New Job
  </h5>
  <div
    className="form-group"
  >
    <label
      className="required"
      htmlFor="job-title"
    >
      Tilte
    </label>
    <input
      className="form-control"
      id="job-title"
      name="title"
      onChange={[Function]}
      placeholder="eg. Administrator"
      required={true}
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="job-location"
    >
      Location
    </label>
    <input
      className="form-control"
      id="job-location"
      name="location"
      onChange={[Function]}
      placeholder="eg. Melbourne"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="job-location"
    >
      Description
    </label>
    <textarea
      className="form-control"
      id="job-location"
      name="description"
      onChange={[Function]}
      placeholder="Please type description here..."
    />
  </div>
  <div
    className="text-right"
  >
    <button
      className="btn btn-primary"
      disabled="disabled"
      type="submit"
    >
      Post
    </button>
  </div>
</form>
`;
